### PR TITLE
Added error dialog when plugin fails to load

### PR DESCRIPTION
**Pull request by britzl**
This will show a popup dialog when an extension plugin fails to load for any reason.

### CODE REVIEW

**Comment by britzl** [comment]
There was a cyclic dependency between workspace.clj and dialogs.clj. In order to break the cycle I had to move the resource dialog code from dialog.clj to new file resource_dialog.clj.

**Comment by vlaaad** [comment]
There is going to be so many merge conflicts between this and #6800 :/

File: editor/src/clj/editor/atlas.clj [hunk 1]
```diff
@@ -40,6 +39,7 @@
             [editor.resource :as resource]
             [editor.resource-io :as resource-io]
             [editor.resource-node :as resource-node]
+            [editor.resource-dialog :as resource-dialog]
```
Comment (matgis) [inline editor/src/clj/editor/atlas.clj:42]: Nit-pick: We try to keep these sorted by name, so this should technically be above `resource-node`.

File: editor/src/clj/editor/workspace.clj [hunk 2]
```diff
@@ -274,11 +268,20 @@ ordinary paths."
...
+        (log/info :msg (str "Loaded plugin " (resource/path resource)))))
+    (log/info :msg (str "Unable to load plugin " (resource/path resource)))
+    (catch Exception e
+      (log/error :msg (str "Exception while loading plugin: " (.getMessage e)))
+      (ui/run-later
+        (do (dialogs/make-info-dialog
+           {:title "Unable to Load Plugin"
+            :icon :icon/triangle-error
+            :header (format "The editor plugin '%s' is not compatible with this version of the editor. Please edit your project dependencies to refer to a suitable version." (resource/proj-path resource))})
+             false)))))
```
Comment (matgis) [inline editor/src/clj/editor/workspace.clj:284]: The indentation is a bit messed up here.
```suggestion
        (do
          (dialogs/make-info-dialog
            {:title "Unable to Load Plugin"
             :icon :icon/triangle-error
             :header (format "The editor plugin '%s' is not compatible with this version of the editor. Please edit your project dependencies to refer to a suitable version." (resource/proj-path resource))})
          false)))))
```

**Review by matgis — APPROVED** [review]
Looks great!

**Comment by vlaaad** [comment]
Maybe I'm missing something, but I think we can achieve a significantly smaller diff if we move `editor.workspace/resource-icon` fn to `editor.resource` namespace, then there will be no need to move select list dialog and resource dialog to another ns at all?

**Comment by matgis** [comment]
> Maybe I'm missing something, but I think we can achieve a significantly smaller diff if we move `editor.workspace/resource-icon` fn to `editor.resource` namespace, then there will be no need to move select list dialog and resource dialog to another ns at all?

Yes, there's more to it. `dialog.clj` also indirectly referred to `workspace.clj` by requiring `editor.defold-project` and `editor.icons`. I've send you a link to the thread where we discussed this on Slack. I think we'll have to tank the potiential merge conflict this time. 😞

**Comment by britzl** [comment]
With the updates from editor-dev this got even messier. Had to split up dialogs.clj a bit more. Not happy about the result...

**Comment by vlaaad** [comment]
After merging #6800 I fixed this PR, it should be superseded by #6834

**Comment by britzl** [comment]
> After merging #6800 I fixed this PR, it should be superseded by #6834

Thanks!

**Pull request closed without merging**